src/app/clusters/webrtc-transport-requestor-server/webrtc-transport-requestor-cluster.cpp

```diff
@@ -0,0 +1,355 @@
+    // If it doesn't match, return false
+    return (peerNodeId == existingSession->peerNodeID) && (peerFabricIndex == existingSession->GetFabricIndex());
+}
+
+DataModel::ActionReturnStatus WebRTCTransportRequestorServer::HandleOffer(const Commands::Offer::DecodableType & req,
+                                                                          const CommandHandler & commandHandler)
+{
+    uint16_t sessionId = req.webRTCSessionID;
+
+    if (!IsPeerNodeSessionValid(sessionId, commandHandler))
+    {
+        return Protocols::InteractionModel::Status::NotFound;
+    }
+
+    // Create arguments for WebRTCTransportRequestorDelegate.
+    WebRTCTransportRequestorDelegate::OfferArgs args;
+    args.sdp = std::string(req.sdp.data(), req.sdp.size());
+
+    // Convert ICE servers list from DecodableList to vector.
+    if (req.ICEServers.HasValue())
+    {
+        auto iterator = req.ICEServers.Value().begin();
+        while (iterator.Next())
+        {
+            args.iceServers.Value().push_back(iterator.GetValue());
+        }
+
+        // Check the list validity
+        CHIP_ERROR err = iterator.GetStatus();
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(Zcl, "HandleOffer: ICEServers list error: %" CHIP_ERROR_FORMAT, err.Format());
+            return Protocols::InteractionModel::Status::InvalidCommand;
+        }
+    }
+
+    // Convert ICETransportPolicy from CharSpan to std::string.
+    if (req.ICETransportPolicy.HasValue())
+    {
+        args.iceTransportPolicy.SetValue(std::string(req.ICETransportPolicy.Value().data(), req.ICETransportPolicy.Value().size()));
+    }
+
+    // Delegate processing: handle the SDP offer, gather ICE candidates, SDP answer, etc.
+    return mDelegate.HandleOffer(sessionId, args);
+}
+
+DataModel::ActionReturnStatus WebRTCTransportRequestorServer::HandleAnswer(const Commands::Answer::DecodableType & req,
+                                                                           const CommandHandler & commandHandler)
+{
+    uint16_t sessionId = req.webRTCSessionID;
+    auto sdpSpan       = req.sdp;
+
+    if (!IsPeerNodeSessionValid(sessionId, commandHandler))
+    {
+        return Protocols::InteractionModel::Status::NotFound;
+    }
+
+    std::string sdpAnswer(sdpSpan.data(), sdpSpan.size());
+
+    // Delegate handles Answer command received.
+    return mDelegate.HandleAnswer(sessionId, sdpAnswer);
+}
+
+DataModel::ActionReturnStatus
+WebRTCTransportRequestorServer::HandleICECandidates(const Commands::ICECandidates::DecodableType & req,
+                                                    const CommandHandler & commandHandler)
+{
+    uint16_t sessionId = req.webRTCSessionID;
+
+    std::vector<ICECandidateStruct> candidates;
+    auto iter = req.ICECandidates.begin();
+    while (iter.Next())
+    {
+        // Get current candidate.
+        const ICECandidateStruct & candidate = iter.GetValue();
```

Comment:
Pre-existing, so a followup to fix: this is not enforcing the "min 1" constraint on the SDPMid field of the candidates.

Reply:
https://github.com/project-chip/connectedhomeip/issues/40339